### ISSUE TITLE
Race condition between TP_Flag reading and action sent by the LCD Panel

### ISSUE THREAD

**Issue by andrivet** [andrivet/ADVi3pp#176]
- [x] This issue is **NOT** about support. **This is not the right place to ask for support**. You **may** get support from the [ADVI3++ Facebook page](https://www.facebook.com/advi3pp/). You can get support from the developer of ADVi3++ from its [Patreon page](https://www.patreon.com/andrivet).

------------------------------------------------------------
### Bug Report

- [x] You have checked that you are not creating a **duplicate** of an existing (maybe closed) issue (use the search)
- [x] You have tested with the **latest** version of ADVi3++ (including Beta releases)
- [x] You are able to reproduce the bug
- [x] You are able to describe step by step the process to reproduce the bug

* Exact model of your **printer**:
* **ADVi3++** exact version: 4.0.0 Alpha 1
* Printing using **SD card** or **USB**: n/a

#### Description of the bug

* Expected behaviour: Reading the TP_Flag state
* Actual behaviour: `### ERROR: Command in response (83) does not correspond to request (81)`
* Steps to reproduce:
  - Start PID Tuning
  - Press very quickly 2 times on the Back button
  - The logs displays:

```
### ERROR: Command in response (83) does not correspond to request (81)
### ERROR: Reading TouchPanelFlag
Auto PID finished: failed
echo:enqueueing "M106 S0"
Silent Buzz
Try to read a word after the end of data
=R=> 5A words, Action = 0x501, KeyValue = 0x0
```

I suspect a race condition:
1. Request the value of the TP_Flag
2. Action sent by the LCD Panel
3. The firmware believes it is the answer of 1.

**Comment by andrivet** [comment]
This bug is not entirely fixed, even if it is mostly silent in the current version. It manifest itself when, sometimes, pressing a command on the LCD screen is not taken into account the first time. In 3D Labs firmware, it generates sometimes a movement on the X axis.

The race condition is the following:
1. A command is sent to the LCD to read the TP_Flag (to know if the screen was touched or not)
2. Exactly at the same time, the user press a command on the screen
3. As a consequence, the response expected by 1. is this command and not the state of the TP_Flag. The response received by 2. is the state of the TP_Flag and not the command.

I experimented different solutions, such as:
* Let the screen do the dimming (by using the TP_LED flag of the R2 register) instead of doing the dimming manually (by reading the TP_Flag). But as R6 is not writable at runtime (I tried), we loose the capability to control the brightness of the screen. It is possible to change the brightness but it is forget when the screen is dimmed.
* Lower the frequency of the TP_Flag reading. It decrease the problem but does not fix it.

A proper solution is probably to refactor the way the firmware interacts with the LCD panel.
